pkg/rpc/context_test.go

```diff
@@ -353,7 +368,15 @@ func TestHeartbeatHealthTransport(t *testing.T) {
 	// Should stay unhealthy despite reconnection attempts.
 	errUnhealthy := errors.New("connection is still unhealthy")
 	if err := util.RetryForDuration(100*clientCtx.heartbeatInterval, func() error {
```

Comment:
It's somewhat unrelated to your fix, but this retry loop looks pretty silly. `RetryForDuration` keeps retrying until a nil error is returned to it or it times out. If it times out, it only returns the last error it received to its caller. Because this closure never returns a nil error, we're effectively calling it many times over the course of `100*clientCtx.heartbeatInterval`, but only ever examining the output of the last call.